### PR TITLE
Adding Encryption Rekey Feature.

### PR DESCRIPTION
**Pull request by anathoodell**
# Description
A few sentences describing the overall goals of the pull request's commits.

# GitHub Issues
List the GitHub issues impacted by this PR:

| GitHub Issue # |
| -------------- |
| |

# Checklist:

- [X] Have you run a grammar and spell checks against your submission?
- [X] Have you tested the changes locally?
- [X] Have you tested whether the hyperlinks are working properly?
- [ ] Did you add the examples wherever applicable?
- [ ] Have you added high-resolution images?

### CODE REVIEW

File: content/docs/secure/encryption/_index.md [hunk 1]
```diff
@@ -117,6 +117,10 @@ Access to the data will be lost for ever.
 
 Refer to [Vault Configuration section](vault) for minimal configuration steps required to support Encryption and other configuration considerations.
 
+## Key Rotation (rekey)
+This preview of encryption includes the ability to change the KEK of an encrypted volume, an operation commonly known as Shallow Rekey, or
```
Comment (AronAtDell) [inline content/docs/secure/encryption/_index.md:121]: I would put "Key Encryption Key (KEK)" for the first occurrence and then use KEK in the rest of the documentation.

**Review by rsedlock1958 — DISMISSED** [review]
I have reviewed this. No edits are required.

File: content/docs/secure/encryption/_index.md (hunk 1 above)
Comment (anathoodell) [reply content/docs/secure/encryption/_index.md:121]: sounds good.

**Comment by anathoodell** [comment]
> I have reviewed this. No edits are required.

please review rekey.md (new file)

**Review by rsedlock1958 — APPROVED** [review]
I have reviewed this. No edits are required.

File: content/docs/secure/encryption/_index.md (hunk 1 above)
Comment (anathoodell) [reply content/docs/secure/encryption/_index.md]: @AronAtDell I have addressed this change in the latest edits. Please review rekey.md as this file is new and contains the brunt of the changes for rekey documentation. Thank you.

File: content/docs/secure/encryption/rekey.md [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+imagePullPolicy: IfNotPresent
+
+# This value is required and must match encryption.pluginName value
+# of the corresponding Dell CSI driver.
+provisioner: sec-isilon.dellemc.com
+# This represents the port number used by encryption driver Rest API server.
+port: 3838
+```
+> **NOTE**: It is important that the `provisioner` reflect the same name as the `pluginName` specified in the standard encryption values.yaml file, and the `port`
+reflect the same port number as the encryption driver REST API server would be listening on. If they don't match then all rekey attempts will fail.
```
Comment (alexemc) [inline content/docs/secure/encryption/rekey.md:41]: Instead of this, standard practice in csm-docs is to place a table with Parameter,Description,Required,Default. 
For `provisioner` the description should be just: "This value is required and must match encryption.pluginName value of the corresponding Dell CSI driver."
For `port` the description should be just: "This value is required and must match encryption.apiPort value of the corresponding Dell CSI driver."
image, imagePullPolicy and logLevel - not required and have default values
provisioner and port - required and have no default values

File: content/docs/secure/encryption/rekey.md [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+
+## Deploy Rekey Controller
+
+Copy the values.yaml to a local file. Once the CSM encryption Rekey Controller local values.yaml file has been
+adjusted for the current cluster, deploy the controller by installing
+the helm chart. As an example:
```
Comment (alexemc) [inline content/docs/secure/encryption/rekey.md:47]: "Helm"

File: content/docs/secure/encryption/rekey.md [hunk 4]
```diff
@@ -0,0 +1,177 @@
...
+If dellctl CLI is installed, rekey'ing a volume is simple. First, identify a volume with a PV that is encrypted with the CSM encryption driver
+provisioner. Verify that there is at least one volume attachment for this PV.
+
+For example, let's rekey an encrypted PV with the name ``K8s-asdf1234``:
+
+``dellctl encryption rekey myrekey k8s-asdf1234``
+
+```
+INFO rekey request "myrekey2" submitted successfully for persistent volume "k8s-112a5d41bc".
+INFO Run 'dellctl encryption rekey-status myrekey2' for more details.
```
Comment (alexemc) [inline content/docs/secure/encryption/rekey.md:71]: The rekey command was for "myrekey", not "myrekey2". PV name is different too.

File: content/docs/secure/encryption/rekey.md [hunk 5]
```diff
@@ -0,0 +1,177 @@
...
+```
+
+Then to check the status of the newly created rekey with the name ``myrekey``:
+
+``dellctl encryption rekey-status myrekey``
+
+```INFO Status of rekey request myrekey2 = completed```
+
+See below for possible Status values and explanations.
```
Comment (alexemc) [inline content/docs/secure/encryption/rekey.md:80]: Since it's not directly below, it would be very nice to put a link to the exact location.

File: content/docs/secure/encryption/rekey.md [hunk 6]
```diff
@@ -0,0 +1,177 @@
...
+
+Now create a Rekey Custom Resource to start the rekey of a volume. For example, there is a PV with a name
+`k8s-09a76734f` as follows:
+```
+Name:            k8s-029a76734f
+Labels:          <none>
+Annotations:     pv.kubernetes.io/provisioned-by: sec-isilon.dellemc.com
+...
+Events:                <none>
+```
```
Comment (alexemc) [inline content/docs/secure/encryption/rekey.md:95]: Why is this needed?

File: content/docs/secure/encryption/rekey.md [hunk 7]
```diff
@@ -0,0 +1,177 @@
...
+    Subresource:     status
+    Time:            2022-11-11T14:01:51Z
+  Resource Version:  21062146
+  UID:               388a0245-18be-4e35-8618-814f30960b43
+Spec:
+  Persistent Volume Name:  k8s-029a76734f
+Status:
+  Phase:  completed
+Events:   <none>
+```
```
Comment (alexemc) [inline content/docs/secure/encryption/rekey.md:154]: Replace this with `kubectl get` which is less verbose:
```
# kubectl get rekey rekey4 -o yaml
apiVersion: encryption.storage.dell.com/v1alpha1
kind: Rekey
metadata:
  annotations:
    kubectl.kubernetes.io/last-applied-configuration: |
      {"apiVersion":"encryption.storage.dell.com/v1alpha1","kind":"Rekey","metadata":{"annotations":{},"name":"rekey4"},"spec":{"persistentVolumeName":"k8s-575b291c63"}}
  creationTimestamp: "2022-11-15T18:25:32Z"
  generation: 1
  name: rekey4
  resourceVersion: "23534992"
  uid: 81a85828-9777-4ffa-a25e-51a4416ff871
spec:
  persistentVolumeName: k8s-575b291c63
status:
  phase: rejected
```

Or better yet, just use this command to get the status:
```
echo $(kubectl get rekey rekey4 -o jsonpath='{.status.phase}')
```

File: content/docs/secure/encryption/rekey.md [hunk 8]
```diff
@@ -0,0 +1,177 @@
...
+The `Status.Phase` field can have the following possible values:
+
+| Status.Phase    | **Description** |
+|-----------------| ----------- | 
+| **initialized** | Preconditions for Rekeying the volume have been met, the volume is attached and encrypted with the CSM encryption driver provisioner.
+| **started**     | The Rekey process has been started with the encryption driver, a connection has been attempted, and a response is expected shortly.
+| **unknown**     | The Rekey process suffered a communication failure while waiting for a response from the encryption driver, and it is possible that the rekey was successfull.
+| **failed**      | The Rekey process has failed, possibly due to a lack of a volume attachment, or no encryption driver could be reached to perform the rekey.
+| **rejected**    | It is not possible to Rekey this volume either because the volume is not encrypted with the specified encryption driver provisioner, or there was no such PV for this volume, or no volume attachments were found.
+| **completed**   | The Rekey successfully completed, which directly implies the KEK of the volume has been changed successfully.
```
Comment (alexemc) [inline content/docs/secure/encryption/rekey.md:170]: I suggest removing majority of the details explaining the internals of the process. Also, users don't have to deal with and know about volume attachments to do rekeying. Please change to a more high level description.

File: content/docs/secure/encryption/troubleshooting.md [hunk 9]
```diff
@@ -85,3 +85,18 @@ Resolution:
 - check the license is for the cluster on which the encrypted volumes are created
 - check [encryption-license secret](../deployment#secret-encryption-license)
 
+# Typical Rekey Failure reasons
+If all rekeys in the cluster are failing 
+- Check the Rekey controller helm chart values.yaml `provisioner` name against the encryption chart `pluginName`, and ensure they match.
+- Check the Rekey controller helm chart values.yaml `port` number against the encryption chart rekey port number, and ensure they match.
```
Comment (alexemc) [inline content/docs/secure/encryption/troubleshooting.md:91]: Check the Rekey Controller Helm chart `port` value against related Dell CSI driver chart `encryption.apiPort` value.

File: content/docs/secure/encryption/troubleshooting.md [hunk 10]
```diff
@@ -85,3 +85,18 @@ Resolution:
 - check the license is for the cluster on which the encrypted volumes are created
 - check [encryption-license secret](../deployment#secret-encryption-license)
 
+# Typical Rekey Failure reasons
+If all rekeys in the cluster are failing 
+- Check the Rekey controller helm chart values.yaml `provisioner` name against the encryption chart `pluginName`, and ensure they match.
```
Comment (alexemc) [inline content/docs/secure/encryption/troubleshooting.md:90]: Check the Rekey Controller Helm chart `provisioner` value against related Dell CSI driver chart `encryption.pluginName` value.

File: content/docs/secure/encryption/rekey.md [hunk 11]
```diff
@@ -0,0 +1,177 @@
+---
+title: "Rekey Configuration"
+linkTitle: "Rekey Configuration"
+weight: 3
```
Comment (alexemc) [inline content/docs/secure/encryption/rekey.md:4]: Please update weight for all pages:
deplyment - 1
uninstallation - 2
vault - 3
rekey - 4
troubleshooting -5
release - 6

File: content/docs/secure/encryption/rekey.md [hunk 12]
```diff
@@ -0,0 +1,177 @@
...
+> **NOTE**: It is important that the `provisioner` reflect the same name as the `pluginName` specified in the standard encryption values.yaml file, and the `port`
+reflect the same port number as the encryption driver REST API server would be listening on. If they don't match then all rekey attempts will fail.
+
+## Deploy Rekey Controller
+
+Copy the values.yaml to a local file. Once the CSM encryption Rekey Controller local values.yaml file has been
+adjusted for the current cluster, deploy the controller by installing
+the helm chart. As an example:
+
+``` helm install --values local-values.yaml isilon-rekey csm-encryption-rekey-controller```
```
[the remark this answers is not in the document]
Comment (anathoodell) [reply content/docs/secure/encryption/rekey.md:49]: that is a much better name.

File: content/docs/secure/encryption/rekey.md [hunk 13]
```diff
@@ -0,0 +1,177 @@
...
+#  IfNotPresent: Only pull the image if it does not already exist on the node.
+#  Never: Never pull the image.
+imagePullPolicy: IfNotPresent
+
+# This value is required and must match encryption.pluginName value
+# of the corresponding Dell CSI driver.
+provisioner: sec-isilon.dellemc.com
+# This represents the port number used by encryption driver Rest API server.
+port: 3838
+```
```
[the remark this answers is not in the document]
Comment (anathoodell) [reply content/docs/secure/encryption/rekey.md:39]: fixed.

File: content/docs/secure/encryption/rekey.md (hunk 12 above)
[the remark this answers is not in the document]
Comment (alexemc) [reply content/docs/secure/encryption/rekey.md:49]: Thinking about it, this is not a powerscale specific instruction, so it's better to name it just `rekey-controller`.

File: content/docs/secure/encryption/deployment.md [hunk 14]
```diff
@@ -37,12 +43,22 @@ encryption:
...
+  logLevel: "debug"
+
   # livenessPort: HTTP liveness probe port number. 
   # Leave empty to disable the liveness probe.
   # Example: 8080
   livenessPort:
 
+  # ocp: Enable when running on OpenShift Container Platform with CoreOS worker nodes.
+  ocp: false
+
```
[the remark this answers is not in the document]
Comment (anathoodell) [reply content/docs/secure/encryption/deployment.md:61]: `ocpCoreID` added

File: content/docs/secure/encryption/deployment.md [hunk 15]
```diff
@@ -52,14 +68,16 @@ encryption:
...
+| --------- | ----------- |----------| ------- |  
+| enabled | Enable/disable volume encryption feature. | No       | false |
+| pluginName | The name of the provisioner to use for encrypted volumes. | No       | "sec-isilon.dellemc.com" |
+| image | Encryption driver image name. | No       | "dellemc/csm-encryption:v0.1.0" |
+| imagePullPolicy | If specified, overrides the chart global imagePullPolicy. | No       | CSI driver global imagePullPolicy |
+| logLevel | Log level of the encryption driver.<br/>Allowed values: "error", "warning", "info", "debug, `"trace". | No       | "error" |
+| apiPort | TCP Port number used by the REST API Server, must match with rekey controller | Yes | 3838 |
+| livenessPort | HTTP liveness probe port number. Leave empty to disable the liveness probe. | No       | |
+| ocp | Enable when running an OCP Platform with CoreOS worker nodes | yes      | false |
```
[the remark this answers is not in the document]
Comment (anathoodell) [reply content/docs/secure/encryption/deployment.md:79]: `ocpCoreID` added.

File: content/docs/secure/encryption/deployment.md [hunk 16]
```diff
@@ -52,14 +68,16 @@ encryption:
...
-| extraArgs | Extra command line parameters to pass to the encryption driver.<br/>Allowed values:<br/>"\-\-sharedStorage" - may be required by some applications to work properly.<br/>When set, performance is reduced and hard links cannot be created.<br/>See the [gocryptfs documentation](https://github.com/rfjakob/gocryptfs/blob/v2.2.1/Documentation/MANPAGE.md#-sharedstorage) for more details. | No | [] |
+| --------- | ----------- |----------| ------- |  
+| enabled | Enable/disable volume encryption feature. | No       | false |
+| pluginName | The name of the provisioner to use for encrypted volumes. | No       | "sec-isilon.dellemc.com" |
+| image | Encryption driver image name. | No       | "dellemc/csm-encryption:v0.1.0" |
+| imagePullPolicy | If specified, overrides the chart global imagePullPolicy. | No       | CSI driver global imagePullPolicy |
+| logLevel | Log level of the encryption driver.<br/>Allowed values: "error", "warning", "info", "debug, `"trace". | No       | "error" |
+| apiPort | TCP Port number used by the REST API Server, must match with rekey controller | Yes | 3838 |
```
[the remark this answers is not in the document]
Comment (anathoodell) [reply content/docs/secure/encryption/deployment.md:77]: sounds good

File: content/docs/secure/encryption/deployment.md [hunk 17]
```diff
@@ -168,3 +186,11 @@ These fields are available for use in *client.json*:
...
 | tls_config.client_key | Set to "/etc/dea/vault/client.key" | Yes | |
 | tls_config.client_ca | Set to "/etc/dea/vault/server-ca.crt" | Yes  | |
+
+### Secret *cluster-kube-config*
+A secret which contains the kube config data needed by the rekey controller for the purposes of rekeying volumes.  Populate this secret with your cluster configuration.
+
+Example kubectl command to create the cluster-kube-config secret:
+```shell
+kubectl create secret generic cluster-kube-config --from-file=/root/.kube/config
+```
```
[the remark this answers is not in the document]
Comment (anathoodell) [reply content/docs/secure/encryption/deployment.md:196]: removed from this page. and placed on rekey controller page.

File: content/docs/secure/encryption/rekey.md [hunk 18]
```diff
@@ -0,0 +1,177 @@
...
+linkTitle: "Rekey Configuration"
+weight: 3
+Description: >
+  Rekey Configuration and Usage
+---
+
+## Rekey Controller Installation
+
+The CSM Encryption Rekey Controller must be installed in order to be able to perform key rotation on CSM encrypted volumes in a
+Kubernetes cluster. The Rekey Controller can be installed via the standard Dell Helm Chart repository available at  https://github.com/dell/helm-charts.git.
```
[the remark this answers is not in the document]
Comment (anathoodell) [reply content/docs/secure/encryption/rekey.md:12]: sounds good.

File: content/docs/secure/encryption/rekey.md [hunk 19]
```diff
@@ -0,0 +1,177 @@
...
+
+A secret of the cluster config must be created with the name ``cluster-kube-config`` typically from the .kube/config. Here is an example:  
+
+```
```
[the remark this answers is not in the document]
Comment (anathoodell) [reply content/docs/secure/encryption/rekey.md:16]: done

File: content/docs/secure/encryption/deployment.md [hunk 20]
```diff
@@ -16,6 +16,12 @@ the rest of the deployment process is described in the correspondent [CSI driver
 Hashicorp Vault must be [pre-configured](../vault) to support Encryption. The Vault server's IP address and port must be accessible 
 from the Kubernetes cluster where the CSI driver is to be deployed.
 
+## Rekey Controller
+
+The CSM Encryption Rekey Controller must be installed in order to be able to perform key rotation on encrypted volumes in a 
+Kubernetes cluster. The Rekey Controller can be installed via the standard Dell Helm Chart repository available at  https://github.com/dell/helm-charts.
```
Comment (alexemc) [inline content/docs/secure/encryption/deployment.md:22]: Please remove once extra space in front of the url.

Comment (anathoodell) [reply content/docs/secure/encryption/deployment.md]: done

File: content/docs/secure/encryption/rekey.md (hunk 11 above)
Comment (anathoodell) [reply content/docs/secure/encryption/rekey.md]: updated

File: content/docs/secure/encryption/troubleshooting.md (hunk 10 above)
Comment (anathoodell) [reply content/docs/secure/encryption/troubleshooting.md]: updated

File: content/docs/secure/encryption/troubleshooting.md (hunk 9 above)
Comment (anathoodell) [reply content/docs/secure/encryption/troubleshooting.md]: fixed.

File: content/docs/secure/encryption/rekey.md (hunk 8 above)
Comment (anathoodell) [reply content/docs/secure/encryption/rekey.md]: simplified the text

File: content/docs/secure/encryption/rekey.md (hunk 7 above)
Comment (anathoodell) [reply content/docs/secure/encryption/rekey.md]: fixed as recommended.

File: content/docs/secure/encryption/rekey.md (hunk 6 above)
Comment (anathoodell) [reply content/docs/secure/encryption/rekey.md]: This was to convey the pv name. Feels redundant in this context, removed.

File: content/docs/secure/encryption/rekey.md (hunk 5 above)
Comment (anathoodell) [reply content/docs/secure/encryption/rekey.md]: added a link.

File: content/docs/secure/encryption/rekey.md (hunk 4 above)
Comment (anathoodell) [reply content/docs/secure/encryption/rekey.md]: fixed the names.

File: content/docs/secure/encryption/rekey.md (hunk 3 above)
Comment (anathoodell) [reply content/docs/secure/encryption/rekey.md]: Updated

File: content/docs/secure/encryption/rekey.md (hunk 12 above)
[the remark this answers is not in the document]
Comment (anathoodell) [reply content/docs/secure/encryption/rekey.md]: replaced with `rekey-controller`

File: content/docs/secure/encryption/rekey.md (hunk 2 above)
Comment (anathoodell) [reply content/docs/secure/encryption/rekey.md]: adopting a table format as recommended.

File: content/docs/secure/encryption/rekey.md [hunk 21]
```diff
@@ -0,0 +1,177 @@
...
+kubectl create secret generic cluster-kube-config --from-file=/home/root/.kube/config
+```
+
+The `csm-encryption-rekey-controller/values.yaml` must be adapted to the kubernetes cluster via the `provisioner` and `port` parameters.
+Here is an example of a typical values.yaml file:
```
[the remark this answers is not in the document]
Comment (anathoodell) [reply content/docs/secure/encryption/rekey.md]: replaced.

File: content/docs/secure/encryption/rekey.md (hunk 18 above)
[the remark this answers is not in the document]
Comment (anathoodell) [reply content/docs/secure/encryption/rekey.md]: updated and removed extra space.

File: content/docs/secure/encryption/deployment.md (hunk 15 above)
[the remark this answers is not in the document]
Comment (anathoodell) [reply content/docs/secure/encryption/deployment.md]: Updated table.

[the remark this answers is not in the document]
Comment (anathoodell) [reply content/docs/secure/encryption/deployment.md]: done.

File: content/docs/secure/encryption/_index.md [hunk 22]
```diff
@@ -117,6 +117,10 @@ Access to the data will be lost for ever.
 
 Refer to [Vault Configuration section](vault) for minimal configuration steps required to support Encryption and other configuration considerations.
 
+## Key Rotation (rekey)
+This preview of encryption includes the ability to change the KEK (Key Encryption Key) of an encrypted volume, an operation commonly known as Shallow Rekey, or
```
[the remark this answers is not in the document]
Comment (anathoodell) [reply content/docs/secure/encryption/_index.md]: Ok, I think got them all.

File: content/docs/secure/encryption/deployment.md [hunk 23]
```diff
@@ -16,6 +16,12 @@ the rest of the deployment process is described in the correspondent [CSI driver
 Hashicorp Vault must be [pre-configured](../vault) to support Encryption. The Vault server's IP address and port must be accessible 
 from the Kubernetes cluster where the CSI driver is to be deployed.
 
+## Rekey Controller
+
+The CSM Encryption Rekey Controller must be installed in order to be able to perform key rotation on encrypted volumes in a 
+Kubernetes cluster. The Rekey Controller can be installed via the standard Dell Helm Chart repository available at  https://github.com/dell/helm-charts.git.
```
[the remark this answers is not in the document]
Comment (anathoodell) [reply content/docs/secure/encryption/deployment.md]: Updated.

File: content/docs/secure/encryption/rekey.md [hunk 24]
```diff
@@ -1,24 +1,25 @@
...
 
+```shell
+ kubectl create secret generic cluster-kube-config --from-file=/home/root/.kube/config
 ```
-kubectl create secret generic cluster-kube-config --from-file=/home/root/.kube/config
-```
 
-The `csm-encryption-rekey-controller/values.yaml` must be adapted to the kubernetes cluster via the `provisioner` and `port` parameters.
-Here is an example of a typical values.yaml file:
+The Rekey Controller Helm chart defines these values `csm-encryption-rekey-controller/values.yaml`:
```
Comment (alexemc) [inline content/docs/secure/encryption/rekey.md:21]: Please remove " `csm-encryption-rekey-controller/values.yaml`". All charts define their values in values.yaml, no need to highlight this. And the directory name may be any.

File: content/docs/secure/encryption/rekey.md [hunk 25]
```diff
@@ -31,143 +32,109 @@ image: dellemc/csm-encryption-rekey-controller:v0.1.0
...
-reflect the same port number as the encryption driver REST API server would be listening on. If they don't match then all rekey attempts will fail.
+
+| Parameter | Description                                                                                               | Required | Default |
+| --------- |-----------------------------------------------------------------------------------------------------------|----------| ------- |  
+| image | Rekey controller image name.                                                                              | No | Yes |
+| imagePullPolicy | Rekey controller image pull policy.                                                                       | No | Yes |
+| logLevel | Log level of the rekey controller.                                                                        | No | Yes |
+| provisioner | This value is required and must match `encryption.pluginName` value of the corresponding Dell CSI driver. | Yes | No |
+| port | This value is required and must match `encryption.apiPort` value of the corresponding Dell CSI driver.    | Yes | No |
```
Comment (alexemc) [inline content/docs/secure/encryption/rekey.md:54]: The Default column should contain the actual default values, if present. E.g. "info" for logLevel, leave empty for port.

File: content/docs/secure/encryption/rekey.md [hunk 26]
```diff
@@ -31,143 +32,109 @@ image: dellemc/csm-encryption-rekey-controller:v0.1.0
...
-Name:            k8s-029a76734f
-Labels:          <none>
-Annotations:     pv.kubernetes.io/provisioned-by: sec-isilon.dellemc.com
-...
-Events:                <none>
-```
-
-An associated example Rekey CR for this volume:
+`k8s-09a76734f`. An associated example Rekey CR for this volume:
 ```
```
Comment (alexemc) [inline content/docs/secure/encryption/rekey.md:99]: Use ```yaml

File: content/docs/secure/encryption/rekey.md [hunk 27]
```diff
@@ -31,143 +32,109 @@ image: dellemc/csm-encryption-rekey-controller:v0.1.0
...
   name: "example-rekey"
 spec:
   persistentVolumeName: "k8s-029a76734f"
-  ```
+```
+
 
 Apply this CR yaml file to start the rekey process:
 
 ```kubectl create -f my-example-rekey.yaml```
```
Comment (alexemc) [inline content/docs/secure/encryption/rekey.md:111]: Use ```shell

File: content/docs/secure/encryption/rekey.md [hunk 28]
```diff
@@ -31,143 +32,109 @@ image: dellemc/csm-encryption-rekey-controller:v0.1.0
...
 
 ### Inspect Status of Rekey
+Once the CR has been created, after some time, the status of the rekey can be
+inspected through the `status.phase` field of the rekey custom resource.
 
 Obtain the status of the rekey:
 
-```kubectl describe rekey my-example-rekey```
-
+```echo $(kubectl get rekey example-rekey -o jsonpath='{.status.phase}')```
```
Comment (alexemc) [inline content/docs/secure/encryption/rekey.md]: Use ```shell

**Review by alexemc — CHANGES_REQUESTED** [review]

File: content/docs/secure/encryption/rekey.md (hunk 25 above)
Comment (anathoodell) [reply content/docs/secure/encryption/rekey.md:54]: lol

File: content/docs/secure/encryption/rekey.md (hunk 28 above)
Comment (anathoodell) [reply content/docs/secure/encryption/rekey.md]: done

File: content/docs/secure/encryption/rekey.md (hunk 27 above)
Comment (anathoodell) [reply content/docs/secure/encryption/rekey.md]: done

File: content/docs/secure/encryption/rekey.md (hunk 26 above)
Comment (anathoodell) [reply content/docs/secure/encryption/rekey.md]: done

File: content/docs/secure/encryption/rekey.md (hunk 25 above)
Comment (anathoodell) [reply content/docs/secure/encryption/rekey.md]: fixed.

File: content/docs/secure/encryption/rekey.md (hunk 24 above)
Comment (anathoodell) [reply content/docs/secure/encryption/rekey.md]: removed.

**Review by sharmilarama — DISMISSED** [review]

File: content/docs/secure/encryption/deployment.md [hunk 29]
```diff
@@ -51,15 +71,18 @@ encryption:
...
+| ocp | Enable when running an OCP Platform with CoreOS worker nodes.                                                                                                                                                                                                                                 | No       | false |
+|ocpCoreID| User ID and group ID of user core on CoreOS worker nodes. Ignored when ocp is set to false.  | No       | "1000:1000" |
+| extraArgs | Extra command line parameters to pass to the encryption driver.<br/>Allowed values:<br/>"\-\-sharedStorage" - may be required by some applications to work properly.<br/>When set, performance is reduced and hard links cannot be created.<br/>See the [gocryptfs documentation](https://github.com/rfjakob/gocryptfs/blob/v2.2.1/Documentation/MANPAGE.md#-sharedstorage) for more details. | No       | [] |
```
Comment (alexemc) [inline content/docs/secure/encryption/deployment.md:85]: Can you make it compact as it was before?
E.g.
```
| --------- | ----------- | -------- | ------- |  
```
instead of
```
| --------- |-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|----------| ------- |
```
Or
```
| enabled | Enable/disable volume encryption feature. | No | false |
```
instead of
```
| enabled | Enable/disable volume encryption feature.                                                                                                                                                                                                                                                                                                                                                     | No       | false |
```
And do the same for other modified tables.

**Review by alexemc — CHANGES_REQUESTED** [review]

**Review by alexemc — DISMISSED** [review]

File: content/docs/secure/encryption/rekey.md [hunk 30]
```diff
@@ -0,0 +1,142 @@
...
+
+### Rekey with dellctl
+
+If dellctl CLI is installed, rekey'ing a volume is simple. First, identify a volume with a PV that is encrypted with the CSM Encryption driver
+provisioner.
+
+For example, let's rekey an encrypted PV with the name ``k8s-112a5d41bc``, and call our rekey object `myrekey`:
+
+```shell
+$ dellctl encryption rekey myrekey k8s-112a5d41bc
```
Comment (tdawe) [inline content/docs/secure/encryption/rekey.md:79]: These commands can also be added to the CLI docs at `content/docs/references/cli/`

**Review by rsedlock1958 — APPROVED** [review]
I have reviewed this. Everything looks fine.

**Review by rsedlock1958 — APPROVED** [review]
I have reviewed this. No edits are required.

File: content/docs/secure/encryption/rekey.md (hunk 30 above)
Comment (anathoodell) [reply content/docs/secure/encryption/rekey.md:79]: done. please review

File: content/docs/references/cli/_index.md [hunk 31]
```diff
@@ -23,6 +23,8 @@ This document outlines all dellctl commands, their intended use, options that ca
 | [dellctl restore create](#dellctl-restore-create) | Restore an application backup |
 | [dellctl restore delete](#dellctl-restore-delete) | Delete application restores |
 | [dellctl restore get](#dellctl-restore-get) | Get application restores |
+| [dellctl encryption rekey](#dellctl-encryption-rekey) | Rekey an encrypted volume |
+| [dellctl encryption rekey status](#dellctl-encryption-rekey-status) | Get status of a encryption rekey operation |
```
Comment (alexemc) [inline content/docs/references/cli/_index.md:27]: Shouldn't it be `rekey-status`, not `rekey status`?

Comment (anathoodell) [reply content/docs/references/cli/_index.md]: yes. fixed.

File: content/docs/references/cli/_index.md [hunk 32]
```diff
@@ -532,3 +534,46 @@ Get restores with their names
...
+
+
+```
+# dellctl encryption rekey myrekey k8s-5d2cc565d4
+ INFO rekey request "myrekey" submitted successfully for persistent volume "k8s-5d2cc565d4".
+ INFO Run 'dellctl encryption rekey-status myrekey' for more details.
+```
+
+
+### dellctl encryption rekey status
```
Comment (alexemc) [inline content/docs/references/cli/_index.md:561]: `rekey-status`?

**Review by alexemc — APPROVED** [review]

**Review by bpjain2004 — APPROVED** [review]

File: content/docs/secure/encryption/rekey.md [hunk 33]
```diff
@@ -0,0 +1,142 @@
...
+| logLevel | Log level of the rekey controller. | No | info |
+| provisioner | This value is required and must match `encryption.pluginName` value of the corresponding Dell CSI driver. | Yes |  |
+| port | This value is required and must match `encryption.apiPort` value of the corresponding Dell CSI driver. | Yes |  |
+
+## Deploy Rekey Controller
+
+Copy the values.yaml to a local file. Once the CSM Encryption Rekey Controller local values.yaml file has been
+adjusted for the current cluster, deploy the controller by installing the Helm chart. As an example:
+
+``` helm install --values local-values.yaml rekey-controller csm-encryption-rekey-controller```
```
Comment (tdawe) [inline content/docs/secure/encryption/rekey.md:61]: If this is installing the helm chart from our official Dell helm repository, then you can also include the step to add the repository similar to step 3 here (`helm repo add dell https://dell.github.io/helm-charts`): https://dell.github.io/csm-docs/docs/observability/deployment/helm/#install-the-csm-for-observability-helm-chart

And the helm install will need to prefix the chart name with `dell`:
```helm install --values local-values.yaml rekey-controller dell/csm-encryption-rekey-controller```

**Review by tdawe — APPROVED** [review]

**Review by alexemc — APPROVED** [review]

**Review by rsedlock1958 — APPROVED** [review]
I have reviewed this. No edits are required.

**Comment by anathoodell** [comment]
Hi,

Would it possible for you to review this one again? The most recent changes are minor.

Thanks

-Aly

**Review by rsedlock1958 — COMMENTED** [review]
I have reviewed this. No edits are required.
